src/cmd/services/m3dbnode/main/main_test.go

```diff
@@ -616,3 +634,42 @@ db:
                   endpoint: {{.InitialClusterEndpoint}}
 `
 )
+
+// waitUntilAllShardsAreAvailable continually polls the session checking to see if the topology.Map
+// that the session is currently storing contains a non-zero number of host shard sets, and if so,
+// makes sure that all their shard states are Available.
+func waitUntilAllShardsAreAvailable(t *testing.T, session client.AdminSession) {
+	for {
+		topoMap, err := session.TopologyMap()
+		require.NoError(t, err)
+
+		var (
+			hostShardSets         = topoMap.HostShardSets()
+			allShardsAreAvailable = true
+		)
+
+		if len(hostShardSets) == 0 {
+			// We haven't received an actual topology yet.
+			continue
```

Comment:
Will this be an issue during the turn-up phase of a cluster? Not sure.

Reply:
This is test code only, no effect on production workloads.

Reply:
Or did you mean do we need to worry about this during turn up? If thats your question, not really. All this means is that there will be a (probably less than a second) window between when all the nodes have finished bootstrapping, and any connected clients are made aware that the nodes are queryable. Only really matters in this test case where everything happens so quickly

Reply:
sorry my question was silly and the packages are separated by `_test`. Sorry for the noise